web/src/lib/app.tsx

```diff
@@ -108,76 +102,11 @@ export default class App {
    */
   private bootstrapIOS() {
     document.body.classList.add('ios');
-    // this.renderDebugBox();
-  }
-
-  /**
-   * Get the page name from the url.
-   */
-  private getPageName(href?: string): string {
-    if (!href) {
-      href = window.location.href;
-    }
-    let link = document.createElement('a');
-    link.href = href;
-
-    // Workaround for IE bug where pathname was not prefixed by '/'
-    const pathname = link.pathname;
-    if (pathname.indexOf('/') !== 0) {
-      return '/' + pathname;
-    }
-    return pathname;
   }
 
-  /**
-   * Update the current page based on new url.
-   */
-  private handleNavigation(href: string) {
-    let page = this.getPageName(href);
-    window.history.pushState(null, '', page);
-    this.renderCurrentPage();
-  }
-
-  private renderDebugBox() {
-    render(<DebugBox />, document.body);
-  }
-
-  private renderSpinner() {
-    render(
-      <div id="spinner">
-        <span
-          ref={el => {
-            if (this.progressMeter) {
-              return;
-            }
-
-            this.progressMeter = el as HTMLSpanElement;
-          }}
-        />
-      </div>,
-      document.body,
-      document.getElementById('spinner')
-    );
-  }
-
-  private renderCurrentPage() {
-    // Render the main controller, Pages.
-    render(
-      <Pages
-        user={this.user}
-        api={this.api}
-        navigate={this.handleNavigation}
-        currentPage={this.getPageName()}
-      />,
-      document.body,
-      document.getElementById('main')
-    );
-  }
-
-  async init(): Promise<void> {
-    // Force page to be ready after a specified time, unless pre-loading images finishes first.
+  async componentDidMount() {
     await Promise.race([
-      sleep(LOAD_TIMEOUT),
+      sleep(LOAD_DELAY),
```

Comment:
Oh, I think I meant to do it the other way around, since from my understanding `LOAD_DELAY` was only used to play the fade-in animation, which we do through CSS now. I just reverted it (so no more `LOAD_DELAY` now, but we got `LOAD_TIMEOUT` back). Pardon moi 🙈